src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2760,9 +2759,6 @@ void InstanceKlass::restore_unshareable_info(ClassLoaderData* loader_data, Handl
   constants()->restore_unshareable_info(CHECK);
 
   if (array_klasses() != nullptr) {
-    // To get a consistent list of classes we need MultiArray_lock to ensure
-    // array classes aren't observed while they are being restored.
-    MutexLocker ml(MultiArray_lock);
```

Comment:
Doesn't this revert JDK-8274338?

Reply:
You're right. I didn't believe my own comment when I removed this.  Thank you for pointing out the bug.